### PR TITLE
Fix parallel logic for layered extremum

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #24682

### CODE REVIEW

**Comment by GiudGiud** [comment]
still wrong

File: framework/src/userobjects/LayeredExtremumMaterialProperty.C [hunk 1]
```diff
@@ -64,14 +64,40 @@ LayeredExtremumMaterialProperty::extreme_value(const Real a, const Real b) const
...
+  }
 }
 
 void
 LayeredExtremumMaterialProperty::threadJoin(const UserObject & y)
 {
-  ElementExtremeMaterialProperty::threadJoin(y);
-  LayeredBase::threadJoin(y);
+  const LayeredExtremumMaterialProperty & lb =
+      dynamic_cast<const LayeredExtremumMaterialProperty &>(y);
```
Comment (lindsayad) [inline framework/src/userobjects/LayeredExtremumMaterialProperty.C:97]:
```suggestion
      static_cast<const LayeredExtremumMaterialProperty &>(y);
```

File: framework/src/userobjects/LayeredExtremumMaterialProperty.C [hunk 2]
```diff
@@ -64,14 +64,40 @@ LayeredExtremumMaterialProperty::extreme_value(const Real a, const Real b) const
...
+  for (unsigned int i = 0; i < _num_layers; i++)
```
Comment (lindsayad) [inline framework/src/userobjects/LayeredExtremumMaterialProperty.C:98]:
```suggestion
  for (const auto i : make_range(_num_layers))
```

**Review by lindsayad — DISMISSED** [review]

**Comment by GiudGiud** [comment]
Thanks for the review. i ll merge and port your comments on a catch all PR. This is pasted boilerplate code so I can fix it in many places

**Comment by GiudGiud** [comment]
Test failure is in contact module and unrelated

**Comment by GiudGiud** [comment]
@loganharbour need unlock to merge this fix

**Comment by GiudGiud** [comment]
Ended up applying the review and propagating to LayeredBase

**Comment by loganharbour** [comment]
Any luck?

**Review by loganharbour — APPROVED** [review]

**Comment by GiudGiud** [comment]
lemme squash this, dont want to break my own rule

**Review by loganharbour — APPROVED** [review]

**Comment by GiudGiud** [comment]
oh interesting

squashing two commits did not dismiss the review

**Pull request merged**